ui/docs/reference/core/index/README.md

```diff
@@ -1,56 +1,83 @@
 # Index Service
 
-The task _index_ provides a service that indexes successfully completed tasks.
-To get a task indexed you must add routes on the form `index.<namespace>`, where
-`<namespace>` is a dot separated hierarchy **without any slashes**.
+The index service provides a service that indexes successfully completed tasks.
 
-**Example**, see the example below for how to specify routes and keys for
-indexing.
+Tasks are indexed either based on pulse messages from the queue with routing keys of the format `route.index.<namespace>` or (less frequently) via calls to the `insertTask` API method.
 
-```js
-{
-  payload:  { /* ... */ },
-  routes: [
-    // index.<namespace> prefixed routes, tasks CC'ed such a route will be indexed
-    // under the given namespace (note, that the namespace may contain spaces)
-    "index.hg-mozilla-org.mozilla-central.nightly.linux64.debug-build",
-    "index.hg.<revision>.nightly.linux64.debug-build"
-  ],
-  extra: {
-    // Optional details for indexing service
-    index: {
-      // Ordering, this taskId will overwrite any thing that has
-      // rank <= 4000, if not provided zero will always assumed and
-      // paths will be overwritten with latest taskId and data if the were also
-      // set with rank zero.
-      rank:       4000,
-
-      // Specified when the entries expires. max 1 year, defaults to 1 year if not
-      // provided
-      expires:          new Date().toJSON(),
-
-      // A little informal data to store along with taskId
-      data: {
-        hgRevision: "...",
-        commitMessae: "...",
-        whatever...
-      }
-    },
-    // Extra properties for other services...
-  }
-  // Other task properties...
-}
-```
+## Index Hierarchy
+
+Tasks are indexed in a dot (`.`) separated hierarchy called a namespace.
+For example, a task could be indexed with the index path `some-app.<revision>.linux-64.release-build`.
+In this case the following namespaces are created.
+
+ 1. `some-app`,
+ 1. `some-app.<revision>`, and,
+ 1. `some-app.<revision>.linux-64`
```

Comment:
Wow, I didn't know multiple `1.` becomes `1. 2. 3.`.

Reply:
Markdown is amazing!